logins-swift/Sync15Passwords/Sync15Logins/LoginRecord.swift

```diff
@@ -109,41 +109,74 @@ open class LoginRecord {
     
     open func toJSON() throws -> String {
         // We need a String to pass back to rust.
-        let data: Data = try JSONSerialization.data(self.toJSONDict())
+        let data: Data = try JSONSerialization.data(withJSONObject: self.toJSONDict())
         return String(data: data, encoding: String.Encoding.utf8)!
     }
     
     // TODO: handle errors in these... (they shouldn't ever happen
     // outside of bugs since we write the json in rust, but still)
 
-    open init(fromJSONDict dict: [String: Any]) {
+    public convenience init(fromJSONDict dict: [String: Any]) {
         self.init(
-            id: dict["id"]! as String,
-            password: dict["password"]! as String,
-            hostname: dict["hostname"]! as String,
+            id: dict["id"] as? String ?? "",
+            password: dict["password"] as? String ?? "",
+            hostname: dict["hostname"] as? String ?? "",
 
-            username: dict["username"] as? String?,
+            username: dict["username"] as? String ?? "",
 
-            formSubmitURL: dict["formSubmitURL"] as? String?,
-            httpRealm: dict["httpRealm"] as? String?,
+            formSubmitURL: dict["formSubmitURL"] as? String ?? "",
+            httpRealm: dict["httpRealm"] as? String ?? "",
 
-            timesUsed: (dict["timesUsed"] as? Int?) ?? 0,
-            timeLastUsed: (dict["timeLastUsed"] as? Int64?) ?? 0,
-            timeCreated: (dict["timeCreated"] as? Int64?) ?? 0,
-            timePasswordChanged: (dict["timePasswordChanged"] as? Int64?) ?? 0,
+            timesUsed: (dict["timesUsed"] as? Int) ?? 0,
+            timeLastUsed: (dict["timeLastUsed"] as? Int64) ?? 0,
+            timeCreated: (dict["timeCreated"] as? Int64) ?? 0,
+            timePasswordChanged: (dict["timePasswordChanged"] as? Int64) ?? 0,
 
-            usernameField: dict["usernameField"] as? String?,
-            passwordField: dict["passwordField"] as? String?
+            usernameField: dict["usernameField"] as? String ?? "",
+            passwordField: dict["passwordField"] as? String ?? ""
         )
     }
 
-    open init(fromJSONString json: String) throws {
-        let dict = try JSONSerialization.jsonObject(with: json, options: []) as [String: Any];
+
+
+    init(id: String,
```

Comment:
In practice I think a lot of these should be optional, Particularly usernameField, passwordField, etc, but also the timestamps.

Maybe it doesn't matter though.